packages/core/register/src/hook.js

```diff
@@ -0,0 +1,77 @@
+import Parcel from '@parcel/core';
+import {addHook} from 'pirates';
+import process from 'process';
+import path from 'path';
+import fs from 'fs';
+import syncPromise from '@parcel/utils/lib/syncPromise';
+
+let revert = null;
+const DEFAULT_CLI_OPTS = {
+  watch: false
+};
+
+// The filehandler returns a function that transforms the code
+function fileHandler(opts) {
+  let parcel = new Parcel({
+    entries: [path.join(process.cwd(), 'index.js')],
+    cliOpts: opts
+  });
+
+  return function(code, filename) {
+    let isInMonorepo =
+      filename.includes('parcel/packages') &&
+      !filename.includes('core/register');
+    if (filename.includes('@parcel') || isInMonorepo) {
+      return code;
+    }
+
+    try {
+      let result = syncPromise(
+        parcel.runTransform({
+          filePath: filename,
+          env: {
+            context: 'node',
+            engines: {}
+          }
+        })
+      );
+
+      if (result.assets && result.assets.length >= 1) {
+        let codePath =
+          (result.assets[0].output && result.assets[0].output.code) ||
+          result.assets[0].code ||
+          '';
+
+        // Read blobs, replace with cache.readBlobs in the future.
+        let codeContent = fs.readFileSync(
```

Comment:
How would register be able to access the cache store?

I've seen it getting added to the asset at bundling stage in the default-bundler PR, but afaik this isn't written at the runTransform stage.

Reply:
Yeah there is an `asset.getOutput()` function being added in that pr.

Reply:
Awesome, will update it once that PR is merged